modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/Editor.es.js

```diff
@@ -437,6 +452,29 @@ export function Editor({
 		init
 	);
 
+	const InputOutputLength = ({target, url}) => {
+		const {resource} = useResource({
+			fetch,
+			link: location.origin + url,
+			variables: {
+				ddmDataProviderInstanceId: target,
+			},
+		});
+
+		return resource?.inputs?.length + resource?.outputs?.length;
+	};
+
+	const {dataProviderInstanceParameterSettingsURL} = otherProps;
+	let newDataProvider = [];
+
+	newDataProvider = dataProvider?.map((provider) => ({
+		...provider,
+		inputOutputLength: InputOutputLength({
+			target: provider.id,
+			url: dataProviderInstanceParameterSettingsURL,
+		}),
+	}));
```

Comment:
Turn `newDataProvider` into a constant
 ```suggestion
	const newDataProvider = dataProvider?.map((provider) => ({
		...provider,
		inputOutputLength: InputOutputLength({
			target: provider.id,
			url: dataProviderInstanceParameterSettingsURL,
		}),
	})) ?? [];
```